fe/fe-core/src/main/java/com/starrocks/connector/iceberg/ScalarOperatorToIcebergExpr.java

```diff
@@ -249,14 +259,84 @@ private static Object convertBoolLiteralValue(Object literalValue) {
         }
     }
 
-    private static class ExtractLiteralValue extends ScalarOperatorVisitor<Object, Void> {
+    private static class ExtractLiteralValue extends ScalarOperatorVisitor<Object, Type.TypeID> {
+        private boolean needCast(PrimitiveType sourceType, Type.TypeID dstTypeID) {
+            switch (sourceType) {
+                case BOOLEAN:
+                    return dstTypeID != Type.TypeID.BOOLEAN;
+                case TINYINT:
+                case SMALLINT:
+                case INT:
+                    return dstTypeID != Type.TypeID.INTEGER;
+                case BIGINT:
+                    return dstTypeID != Type.TypeID.LONG;
+                case FLOAT:
+                    return dstTypeID != Type.TypeID.FLOAT;
+                case DOUBLE:
+                    return dstTypeID != Type.TypeID.DOUBLE;
+                case DECIMALV2:
+                case DECIMAL32:
+                case DECIMAL64:
+                case DECIMAL128:
+                    return dstTypeID != Type.TypeID.DECIMAL;
+                case HLL:
+                case VARCHAR:
+                case CHAR:
+                    return dstTypeID != Type.TypeID.STRING;
+                case DATE:
+                    return dstTypeID != Type.TypeID.DATE;
+                case DATETIME:
+                    return dstTypeID != Type.TypeID.TIMESTAMP;
+                default:
```

Comment:
ARRAY,MAP,STRUCT？

Reply:
These type usually won't use as partition column, ignore is ok, but i will give a better way.